src/client/datascience/kernel-launcher/localKernelFinder.ts

```diff
@@ -203,14 +201,24 @@ export class LocalKernelFinder implements ILocalKernelFinder {
                     if (
                         kernelspec.language === PYTHON_LANGUAGE &&
                         hideDefaultKernelSpecs &&
-                        kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName)
+                        (kernelspec.name.toLowerCase().match(isDefaultPythonKernelSpecName) ||
+                            kernelspec.argv[0].toLowerCase() === 'python' ||
+                            kernelspec.argv[0].toLowerCase().startsWith('/usr/bin/python'))
```

Comment:
Hide default kernel specs with paths pointing to `/usr/bin/python` (as we can run against these kernels by simply selecting the corresponding interpreter)

Somethign that was missed in https://github.com/microsoft/vscode-jupyter/pull/5732